jumanji/environments/combinatorial/routing/generation_methods_v1_uo.py

```diff
@@ -0,0 +1,572 @@
+import random
+from collections import deque
+from typing import List, Tuple, Union, Optional, Dict, Callable
+import numpy as np
+
+
+class Grid:
```

Comment:
Grid class should maybe be in a separate file since you might be reusing it for other files